### ISSUE TITLE
Trigger runs from GitHub Deployments

### ISSUE THREAD

**Issue by simonbrandhof** [cirruslabs/cirrus-ci-docs#469]
Hi,

Is there already a way to trigger a run when a [GitHub Deployment](https://developer.github.com/v3/repos/deployments/) is created? If not, do you consider this kind of feature for supporting CI/CD pipelines?

Thanks

**Comment by fkorotkov** [comment]
Definitely. I think in case of a deployment Cirrus can specify `$CIRRUS_DEPLOYMENT_ID` environment variable so tasks can be customizable and communicate back overall Build status as a status for the Deployment. 🤔

**Comment by simonbrandhof** [comment]
Yes, that would be already a nice improvement.

Then having a dedicated pipeline, for instance declared in `.cirrus-deployment.yml`, would help to not impacting the configuration of standard build pipeline (avoid `only_if: $CIRRUS_DEPLOYMENT_ID == '' everywhere).

**Comment by fkorotkov** [comment]
Interesting idea about `.cirrus-deployment.yml`. Maybe cirrus can look into `.cirrus` folder and look for `push.yml`, `pr.yml` and `deployment.yml` before defaulting to `.cirrus.yml` in the root of the repository. 🤔

**Comment by RDIL** [comment]
> Interesting idea about `.cirrus-deployment.yml`. Maybe cirrus can look into `.cirrus` folder and look for `push.yml`, `pr.yml` and `deployment.yml` before defaulting to `.cirrus.yml` in the root of the repository. 🤔

I think this would come with templating?
